buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -267,13 +270,30 @@ public void plugin(URI plugin) {
         this.plugins.add(plugin);
     }
 
+    @Override
+    public void plugin(Provider<RegularFile> plugin) {
+        this.plugin(() -> plugin.get().getAsFile().toURI());
+    }
+
+    @Override
+    public void plugin(URI plugin) {
+        this.plugin(() -> plugin);
+    }
+
     @Override
     public void plugin(File plugin) {
-        plugin(plugin.toURI());
+        this.plugin(() -> plugin.toURI());
     }
 
     @Override
     public void module(File module) {
+        RegularFileProperty file = project.getObjects().fileProperty();
+        file.fileValue(module);
+        this.module(file);
+    }
+
+    @Override
+    public void module(Provider<RegularFile> module) {
         this.modules.add(module);
```

Comment:
I think it could simplify some of the other code here if we just made `this.modules` a `List<Provider<File>>` and called `module.getAsFile()` here rather than having to do `getAsFile()` in a bunch of other places.

Reply:
Done.  (not exactly as stated, but the backing list is now a `Provider<File>` and a `module.map(RegularFile::getAsFile)` is used to lazily convert the type.